marlowe-playground-client/src/Marlowe/Gen.purs

```diff
@@ -52,8 +53,8 @@ genRational = do
 genSlot :: forall m. MonadGen m => MonadRec m => m Slot
 genSlot = Slot <$> genBigInteger
 
-genTimeout :: forall m. MonadGen m => MonadRec m => m (TermWrapper Slot)
-genTimeout = TermWrapper <$> genSlot <*> pure zero
+genTimeout :: forall m. MonadGen m => MonadRec m => m H.ExtendedTimeout
+genTimeout = H.Slot <$> genBigInteger
```

Comment:
Maybe for the future but I think we should have tests that exercise holes, template values, the `toCore` function etc which would probably need some more complicated generators (i.e. with a flag). We could maybe have `MonadAsk Env`.

Reply:
100%, not sure what `MonadAsk` is. The hole tests are a bit problematic like with TermWrapper because they do not really do the Blockly roundtrip, we would need to have `ExtendedTimeout` type. The others definitely, I will expand the generators in the future

Reply:
`MonadAsk` is the class of a reader monad (but without the `local` function). There is also `MonadReader` that has `local` and is always a `MonadAsk`.